File: Swift-C++-IR-Comparison/test/structsAndClasses/Swift/Structs/StructReadOnlyComputedProperty.swift
// If a property marked with the lazy modifier is accessed by
// multiple threads simultaneously and the property hasn’t yet
// been initialized, there’s no guarantee that the property
// will be initialized only once.

@main
struct MyMain {

    static func main() {
        let fourByFiveByTwo = Cuboid(width: 4.0, height: 5.0, depth: 2.0)
        let computeMe = fourByFiveByTwo.volume
    }
}
struct Cuboid {
    var width = 0.0, height = 0.0, depth = 0.0
    var volume: Double {
        return width * height * depth
    }
}

File: AutoTranspiledTests/try-0/StructReadOnlyComputedProperty.cpp

struct Cuboid {
    double width = 0.0, height = 0.0, depth = 0.0;
    double volume() {
        return width * height * depth;
    }
};

int main() {
    Cuboid fourByFiveByTwo = { 4.0, 5.0, 2.0 };
    double computeMe = fourByFiveByTwo.volume();
    return 0;
}